modules/uncertainty/R/ensemble.R

```diff
@@ -461,12 +463,12 @@ write.ensemble.configs <- function(defaults, ensemble.samples, settings, model,
                            settings = settings,
                            new.state = new.state[i, ], 
                            new.params = new.params[[i]], #new.params$`646`[[i]] for debugging
-                           inputs =list(met=list(path=inputs$samples[[i]])), 
+                           inputs =list(met=list(path=inputs$met$samples[[i]]),soilinitcond=list(path=inputs$soilinitcond$samples[[i]])), 
```

Comment:
This line of code, which requires the `met` and `soilinitcond` paths inside the `restart` object, doesn't fit the creation of the `restart.list` object in the `sda.enkf_MultiSite` function (see L452).

Reply:
If I understood your question correctly, my latest edits seemed to have already fixed it. Firstly, "soilinitcond" was added in the XML file in "Create_Multi_settings.R". Then "inputs" were generated for all tags under the "ensemble" here: https://github.com/PecanProject/pecan/pull/3406/files#diff-8678ad23f7d92ce4d8c0e81c8a51298c2b7a1ee76759f9c848fd54492f533c9e. Finally, "inputs" was passed to "samples" when t=1 and restart=FALSE (https://github.com/PecanProject/pecan/blob/6cbf62b6cf78e129a938c7a9a0d3258f51d86748/modules/assim.sequential/R/sda.enkf_MultiSite.R#L478), and restart.list when t>1 here: https://github.com/PecanProject/pecan/blob/6cbf62b6cf78e129a938c7a9a0d3258f51d86748/modules/assim.sequential/R/sda.enkf_MultiSite.R#L452. Let me know if you still have concerns.

Reply:
Here you want to pass in the sampled set of inputs, regardless of what variables are sampled, rather than hard coding a specific subset. Not every model even has the concept of soilinitcond. Plus, if you add new inputs you'd have to change these hard coded variables.

Reply:
Fixed.